Assets/HoloToolkit/Utilities/Editor/ConfigureMenu.cs

```diff
@@ -22,34 +23,186 @@ public static void ShowHelp()
+        /// <param name="camera">Camera to update</param>
+        /// <param name="output">Output log</param>
+        private static void SetCameraDefaults(Camera camera, bool isMain, StringBuilder output)
+        {
+            output.AppendLine(string.Format("     Checking camera {0} at {1}", camera.name, camera.transform.position.ToString()));
+
+            if (isMain)
+            {
+                output.Append("     Checking position:");
+                if (camera.transform.position != Vector3.zero)
+                {
+                    camera.transform.position = Vector3.zero;
+                    output.AppendLine("  Updated position");
+                }
+                else
+                {
+                    output.AppendLine("  OK");
+                }
+            }
+            else
+            {
+                output.AppendLine("     Position ignored, not the MainCamera");
+            }
+
+            output.Append("     Checking Near Clip Plane is .85:");
+            if (camera.nearClipPlane != 0.85f)
+            {
+                camera.nearClipPlane = 0.85f;
+                output.AppendLine("  Updated");
+            }
+            else
+            {
+                output.AppendLine("  OK");
+            }
+
+            output.Append("     Checking Clear Flags are Black:");
+            if (camera.clearFlags != CameraClearFlags.Color)
+            {
+                camera.clearFlags = CameraClearFlags.Color;
+                camera.backgroundColor = Color.black;
+                output.AppendLine("  Updated");
+            }
+            else
+            {
+                output.AppendLine("  OK");
+            }
+            output.AppendLine();
+        }
+
+        /// <summary>
+        /// Applies recommended scene settings to the current scenes
+        /// </summary>
+        [MenuItem("HoloToolkit/Configure/Apply HoloLens Scene Settings", priority = 0)]
+        public static void ApplySceneSettings()
+        {
+            var output = new StringBuilder();
+            ApplySceneSettingsInternal(output);
+            EditorUtility.DisplayDialog("Results", output.ToString(), "Ok");
+        }
+
+        private static bool ApplyProjectSettingsInternal(StringBuilder output)
         {
             // Build settings
-            EditorUserBuildSettings.SwitchActiveBuildTarget(BuildTarget.WSAPlayer);
-            EditorUserBuildSettings.wsaSDK = WSASDK.UWP;
-            EditorUserBuildSettings.wsaUWPBuildType = WSAUWPBuildType.D3D;
+            output.AppendLine("** Checking Build Settings for WSA");
+            bool isWsa = EditorUserBuildSettings.activeBuildTarget == BuildTarget.WSAPlayer;
+            bool isWsaSdk = EditorUserBuildSettings.wsaSDK == WSASDK.UWP;
+            bool isDirect3D = EditorUserBuildSettings.wsaUWPBuildType == WSAUWPBuildType.D3D;
+            bool isUnityCSharpProjs = EditorUserBuildSettings.wsaGenerateReferenceProjects;
```

Comment:
I'd totally agree but the sample(s) (fairly recently unless changed)
recommended this in at least one location as such I've seen most people use
it in the field. I don't use it unless necessary (and its caused issues
with a recent unquoted space name) but just following default convention on
this. Most don't realize what it does.

On Sun, Jul 24, 2016 at 12:04 PM, Josh Wittner <notifications@github.com>
wrote:

> In Assets/HoloToolkit/Utilities/Editor/ConfigureMenu.cs
> <https://github.com/Microsoft/HoloToolkit-Unity/pull/134#discussion_r71993518>
> :
>
> >          {
> >              // Build settings
> > -            EditorUserBuildSettings.SwitchActiveBuildTarget(BuildTarget.WSAPlayer);
> > -            EditorUserBuildSettings.wsaSDK = WSASDK.UWP;
> > -            EditorUserBuildSettings.wsaUWPBuildType = WSAUWPBuildType.D3D;
> > +            output.AppendLine("** Checking Build Settings for WSA");
> > +            bool isWsa = EditorUserBuildSettings.activeBuildTarget == BuildTarget.WSAPlayer;
> > +            bool isWsaSdk = EditorUserBuildSettings.wsaSDK == WSASDK.UWP;
> > +            bool isDirect3D = EditorUserBuildSettings.wsaUWPBuildType == WSAUWPBuildType.D3D;
> > +            bool isUnityCSharpProjs = EditorUserBuildSettings.wsaGenerateReferenceProjects;
>
> I also don't think that this option should be a part of this process since
> the generation of reference projects isn't a required or recommended step
> for HoloLens development. Developers should change this option via the
> Build Settings window to suit their needs.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Microsoft/HoloToolkit-Unity/pull/134/files/3933b816638f8cce7b4dc9c87ac4fbcaadee4ded#r71993518>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABqPDUg0OszDIfc7QwCNs8lfRaIg442qks5qY7dEgaJpZM4JTeQs>
> .
>